platform/pulpcore/app/models/content.py

```diff
@@ -79,27 +79,21 @@ class Artifact(Model):
         downloaded (models.BooleanField): The associated file has been successfully downloaded.
         requested (models.BooleanField): The associated file has been requested by a client at
             least once.
-        relative_path (models.TextField): The artifact's path relative to the associated
-            :class:`Content`. This path is incorporated in the absolute storage path of the file
-            and its published path relative to the root publishing directory. At a minimum the path
-            will contain the file name but may also include sub-directories.
         size (models.IntegerField): The size of the file in bytes.
         md5 (models.CharField): The MD5 checksum of the file.
         sha1 (models.CharField): The SHA-1 checksum of the file.
         sha224 (models.CharField): The SHA-224 checksum of the file.
         sha256 (models.CharField): The SHA-256 checksum of the file.
         sha384 (models.CharField): The SHA-384 checksum of the file.
         sha512 (models.CharField): The SHA-512 checksum of the file.
-
-    Relations:
-
-        content (models.ForeignKey): The associated content.
     """
+    def storage_path(self, name):
+        digest = self.sha256
+        return "%s/artifacts/%s/%s" % (settings.MEDIA_ROOT, digest[0:2], digest[2:])
```

Comment:
Also, as we discussed, the storage path should be calculated by something coupled to the actual storage mechanism.  The current implementation makes an attempt at this by having `ArtifactLocation` defined in the _storage_ module but falls a little short of the goal.  I suggest adding a method to the `FileStorage` that can be used to get a _deconstructable_ object that can calculate all paths appropriately for that storage mechanism.   Then in `Artifact.storage_path()` instantiate `DefaultStorage`, get the path calculator and use it to calculate and return the storage path.  We can do the same for the certificate storage in another PR.